common/tracing/src/logging.rs

```diff
@@ -70,125 +58,65 @@ static GLOBAL_UT_LOG_GUARD: Lazy<Arc<Mutex<Option<WorkerGuard>>>> =
-/// The format layer logging span/event in plain text, without color, one event per line.
-/// This is useful in a unit test.
-pub fn init_tracing(app_name: &str, dir: &str, level: &str) -> (WorkerGuard, DefaultGuard) {
-    let (g, sub) = init_file_subscriber(app_name, dir, level);
-
-    let subs_guard = tracing::subscriber::set_default(sub);
-
-    tracing::info!("initialized tracing");
-    (g, subs_guard)
-}
-
-/// Creates a global tracing/logging subscriber which saves events in one log file.
-pub fn init_global_tracing(app_name: &str, dir: &str, level: &str) -> WorkerGuard {
-    let (g, sub) = init_file_subscriber(app_name, dir, level);
-    tracing::subscriber::set_global_default(sub).expect("error setting global tracing subscriber");
-
-    tracing::info!("initialized global tracing");
-    g
-}
-
-/// Create a file based tracing/logging subscriber.
-/// A guard must be held during using the logging.
-/// The format layer logging span/event in plain text, without color, one event per line.
-/// Optionally it adds a layer to send to opentelemetry if env var `DATABEND_JAEGER` is present.
-pub fn init_file_subscriber(
-    app_name: &str,
-    dir: &str,
-    level: &str,
-) -> (WorkerGuard, impl Subscriber) {
-    // open log file
-
-    let f = RollingFileAppender::new(Rotation::HOURLY, dir, app_name);
-
-    // build subscriber
-
-    let (writer, writer_guard) = tracing_appender::non_blocking(f);
-
-    let f_layer = Layer::new()
-        .with_writer(writer)
-        .with_thread_ids(true)
-        .with_thread_names(false)
-        .with_ansi(false)
-        .with_span_events(fmt::format::FmtSpan::FULL);
+    // JSON log layer.
+    let rolling_appender = RollingFileAppender::new(Rotation::HOURLY, dir, app_name);
+    let (rolling_writer, rolling_writer_guard) = tracing_appender::non_blocking(rolling_appender);
+    let file_logging_layer = BunyanFormattingLayer::new(app_name.to_string(), rolling_writer);
+    guards.push(rolling_writer_guard);
+
+    // Span layer.
+    let mut file_span_layer = None;
+    let mut jaeger_layer = None;
+    let fuse_jaeger_env = env::var("DATABEND_JAEGER").unwrap_or_else(|_| "".to_string());
+    if !fuse_jaeger_env.is_empty() {
+        {
+            let span_dir = format!("{}_span", dir);
+            let span_rolling_appender =
+                RollingFileAppender::new(Rotation::HOURLY, span_dir, app_name);
+            let (span_rolling_writer, span_rolling_writer_guard) =
+                tracing_appender::non_blocking(span_rolling_appender);
+            file_span_layer = Some(
+                Layer::new()
+                    .with_writer(span_rolling_writer)
+                    .with_thread_ids(true)
+                    .with_thread_names(false)
+                    .with_ansi(false)
+                    .with_span_events(fmt::format::FmtSpan::FULL),
+            );
+            guards.push(span_rolling_writer_guard);
+        }
```

Comment:
By default metasrv logs events  in the span-format.
In a CI it is by default on and it's saved my `a$$` several times on debugging a distributed service.

Shall we turn the jaeger on for our CI? I am not sure if there are any side effects with jaeger enabled and the bunyun service off, in a CI env.

Otherwise, what about always enabling span logging by default, or enable span logging by setting a different env-var?

Except this, everything else looks great to me :DDD

Reply:
> By default metasrv logs events in the **span-format.**
> In a CI it is by default on and it's saved my a$$ several times on debugging a distributed service.

got it.   let's keep it.

Reply:
I am not sure if we enable log span by default(Like in databend-query), does it affects the performance?

Reply:
At least in non-debug mode, i.e., the `log_level` does not enable a logging, the cost is about the same as an atomic read(`LevelFilter::current()`). 

In a debug mode with log_level = debug, the performance degradation can be perceived by a human. 

The expanded `tracing::debug!()`:
```
{
        if ::tracing::Level::DEBUG <= ::tracing::level_filters::STATIC_MAX_LEVEL
            && ::tracing::Level::DEBUG <= ::tracing::level_filters::LevelFilter::current() {
            use ::tracing::__macro_support::*;
            static CALLSITE: ::tracing::__macro_support::MacroCallsite = {
                use ::tracing::__macro_support::MacroCallsite;
                static META: ::tracing::Metadata<'static> = {
                    ::tracing_core::metadata::Metadata::new(
                        concat!(
                        "event ",
                        file!( ),
                        ":",
                        line!( )
                        ),
                        module_path!( ),
                        ::tracing::Level::DEBUG,
                        Some(file!( )),
                        Some(line!( )),
                        Some(module_path!( )),
                        ::tracing_core::field::FieldSet::new(&[stringify!( message )], ::tracing_core::callsite::Identifier((&CALLSITE))),
                        ::tracing::metadata::Kind::EVENT,
                    )
                };
                MacroCallsite::new(&META)
            };
            let interest = CALLSITE.interest();
            if !interest.is_never() && CALLSITE.is_enabled(interest) {
                let meta = CALLSITE.metadata();

                ::tracing::Event::dispatch(
                    meta,
                    &{
                        #[allow(unused_imports)]
                        use ::tracing::field::{debug, display, Value};
                        let mut iter = (meta.fields()).iter();
                        (meta.fields()).value_set(&[(&(iter.next().expect("FieldSet corrupted (this is a bug)")), Some(&format_args!("id:{} replicate to sm index:{}", self.id, entry.log_id.index
                        ) as &Value))])
                    },
                );
            }
        }
    }
```

Reply:
Ok, I am assuming all the span log in debug level. 
I will enable the span by default later.
Thank you @drmingdrmer .